src/browser/cdp/selectivity/utils.ts

```diff
@@ -0,0 +1,210 @@
+import { sortedIndex, memoize } from "lodash";
+import { SourceMapConsumer, type BasicSourceMapConsumer, type RawSourceMap } from "source-map";
+import fs from "fs";
+import path from "path";
+import { URL } from "url";
+import type { CDPRuntime } from "../domains/runtime";
+import type { CDPSessionId } from "../types";
+import { softFileURLToPath } from "../../../utils/fs";
+import type { NormalizedDependencies } from "./types";
+import { WEBPACK_PROTOCOL } from "./constants";
+
+export const fetchTextWithBrowserFallback = async (
+    url: string,
+    runtime: CDPRuntime,
+    sessionId: CDPSessionId,
+): Promise<string | Error> => {
+    const isSourceMapEmbedded = new URL(url).protocol === "data:";
+
+    if (isSourceMapEmbedded) {
+        // With "data" protocol it just decodes embedded source maps without actual network requests
+        // So we can do it directly from node.js
+        return fetch(url)
+            .then(r => r.text())
+            .catch((err: Error) => err);
+    }
+
+    // At first, trying to fetch sourceMaps directly from the node.js
+    // Then falling back to do it via browser (if Testplane has no direct network access to it, for example)
+    try {
+        return await fetch(url).then(r => r.text());
+    } catch {
+        return runtime
+            .evaluate<string>(sessionId, {
+                expression: `fetch("${url.replaceAll('"', '\\"')}").then(r => r.text())`,
+                awaitPromise: true,
+                returnByValue: true,
+            })
+            .then(r => r.result.value)
+            .catch((err: Error) => err);
+    }
+};
+
+const isWebpackProtocol = (posixRelativeSourceFilePath: string): boolean => {
+    return posixRelativeSourceFilePath.startsWith(WEBPACK_PROTOCOL);
+};
+
+export const patchSourceMapSources = (sourceMap: RawSourceMap, sourceRoot?: string): RawSourceMap => {
```

Comment:
- "source-map" npm package has issue while resolving path with protocol if "sourceRoot" exists
- It also does not know, where webpack source root is
- So we removing "webpack:///" prefix from sources and providing "sourceRoot" from Testplane selectivity config